### PR TITLE
docs(clients): add ACL section to code doc

### PR DESCRIPTION
**Pull request by millotp**
## 🧭 What and Why

As suggested by @morganleroi and @shortcuts, we can add the required ACLs to the clients doc, above each function.
We can't do that by simply adding it to the `{{notes}}` because each language has it's own way to handle multiline comments.

I didn't do java for now, because the javadoc is messed up, will do it in another PR

### CODE REVIEW

**Comment by morganleroi** [comment]
Huuuuuuge

File: templates/csharp/libraries/httpclient/api.mustache [hunk 1]
```diff
@@ -102,7 +102,17 @@ namespace Algolia.Search.Clients
         {{#supportsAsync}}
 
         /// <summary>
-        /// {{{summary}}} {{notes}}
+        /// {{{summary}}}
```
Comment (shortcuts) [inline templates/csharp/libraries/httpclient/api.mustache:105]: maybe just the `description` is enough, IIRC it's required so it removes the redundancy of having both

Comment (shortcuts) [reply templates/csharp/libraries/httpclient/api.mustache:105]: (applies to all languages IMHO)

File: templates/swift/api.mustache [hunk 2]
```diff
@@ -79,7 +79,15 @@ typealias Client = {{classname}}
...
-     {{{.}}}{{/notes}}{{#subresourceOperation}}
+     {{{.}}}{{/notes}}
+     {{#vendorExtensions}}
+     {{#x-acl.0}}
+
+     Required API Key ACLs:{{/x-acl.0}}
+     {{#x-acl}}
+       - {{.}}
+     {{/x-acl}}
+     {{/vendorExtensions}}{{#subresourceOperation}}
```
Comment (shortcuts) [inline templates/swift/api.mustache:90]: this seems to add newlines even when there's no ACLs to display

**Review by shortcuts — COMMENTED** [review]
so nice

File: templates/csharp/libraries/httpclient/api.mustache (hunk 1 above)
Comment (millotp) [reply templates/csharp/libraries/httpclient/api.mustache:105]: there is no description on operations, just summary and notes, maybe I can just leave the notes yes

**Review by Fluf22 — COMMENTED** [review]
I'm not sure that's the best option for Swift, because it removes a useful short description for some operations.
But this can be looked at later!

**Review by Fluf22 — APPROVED** [review]

**Pull request merged**